Restore internal tools dotnet install binds us to only released versions

The [UseDotnet@2 task](https://github.com/dotnet/runtime/blob/363b7add1906547eeba681b3f3ec3f686a603dee/eng/pipelines/common/restore-internal-tools.yml#L6) can only resolve publicly released dotnet SDK versions from https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/5.0/releases.json. This means that we can't update the [sdk attribute in our global.json](https://github.com/dotnet/runtime/blob/363b7add1906547eeba681b3f3ec3f686a603dee/global.json#L3) to a non-officially released version:

```
##[section]Starting: Install dotnet 
==============================================================================
Task         : Use .NET Core
Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
Version      : 2.165.0
Author       : Microsoft Corporation
Help         : https://aka.ms/AA4xgy0
==============================================================================
No matching sdk version could be found for specified version: 5.0.100-preview.3.20170.6 Kindly note the preview versions are only considered in latest version searches if Include Preview Versions checkbox is checked.
Version 5.0.100-preview.3.20170.6 could not be found in its channel, will now search in adjacent channels.
Searching for version in channel 5.0
##[error]sdk version matching: 5.0.100-preview.3.20170.6 could not be found
##[section]Finishing: Install dotnet 
```

We should find another way to aquire the dotnet SDK for internal tools restore. Meanwhile I'm hardcoding the version to the latest officially released version.

cc @dotnet/runtime-infrastructure @safern

--- a/eng/pipelines/common/restore-internal-tools.yml
+++ b/eng/pipelines/common/restore-internal-tools.yml
@@ -1,27 +1,14 @@
-parameters:
-  installDotnet: false
-
 steps:
-  - ${{ if eq(parameters.installDotnet, true) }}:
-    - task: UseDotNet@2
-      displayName: Install dotnet ${{ parameters.dotnetVersion }}
-      inputs:
-        packageType: sdk
-        useGlobalJson: true
-        installationPath: $(Agent.TempDirectory)/_dotnet
-
   - task: NuGetAuthenticate@0
     inputs:
       nuGetServiceConnections: 'dotnet-core-internal-tooling'
       forceReinstallCredentialProvider: true
 
-  - task: DotNetCoreCLI@2
+  - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt)
+            -ci
+            -restore
+            -projects $(Build.SourcesDirectory)/eng/common/internal/Tools.csproj
+            /bl:$(Build.SourcesDirectory)/artifacts/log/$(_BuildConfig)/RestoreInternal.binlog
+            /v:normal
     displayName: Restore internal tools
     condition: and(succeeded(), ne(variables['_skipRestoreInternalTools'], 'true'))
-    inputs:
-      command: restore
-      feedsToUse: config
-      projects: 'eng/common/internal/Tools.csproj'
-      nugetConfigPath: 'eng/internal/NuGet.config'
-      restoreDirectory: '$(Build.SourcesDirectory)\.packages'
-      verbosityRestore: 'diagnostic'

--- a/eng/pipelines/coreclr/templates/build-corelib-job.yml
+++ b/eng/pipelines/coreclr/templates/build-corelib-job.yml
@@ -74,14 +74,12 @@ jobs:
       - script: $(Build.SourcesDirectory)\eng\common\init-tools-native.cmd -InstallDirectory $(Build.SourcesDirectory)\native-tools -Force
         displayName: Install native dependencies
 
-    # # Install internal tools on official builds
-    # # Since our internal tools are behind an authenticated feed,
-    # # we need to use the DotNetCli AzDO task to restore from the feed using a service connection.
-    # # We can't do this from within the build, so we need to do this as a separate step.
+    # Install internal tools on official builds
+    # Since our internal tools are behind an authenticated feed,
+    # we need to use the DotNetCli AzDO task to restore from the feed using a service connection.
+    # We can't do this from within the build, so we need to do this as a separate step.
     - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
       - template: /eng/pipelines/common/restore-internal-tools.yml
-        parameters:
-          installDotnet: true
 
     # Build Private CoreLib
     - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -subset clr.corelib $(crossArg) -arch $(archType) -c $(buildConfig) $(officialBuildIdArg) -ci

--- a/eng/pipelines/coreclr/templates/build-job.yml
+++ b/eng/pipelines/coreclr/templates/build-job.yml
@@ -115,14 +115,12 @@ jobs:
       - script: $(Build.SourcesDirectory)\eng\common\init-tools-native.cmd -InstallDirectory $(Build.SourcesDirectory)\native-tools -Force
         displayName: Install native dependencies
 
-    # # Install internal tools on official builds
-    # # Since our internal tools are behind an authenticated feed,
-    # # we need to use the DotNetCli AzDO task to restore from the feed using a service connection.
-    # # We can't do this from within the build, so we need to do this as a separate step.
+    # Install internal tools on official builds
+    # Since our internal tools are behind an authenticated feed,
+    # we need to use the DotNetCli AzDO task to restore from the feed using a service connection.
+    # We can't do this from within the build, so we need to do this as a separate step.
     - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
       - template: /eng/pipelines/common/restore-internal-tools.yml
-        parameters:
-          installDotnet: true
 
     # Build CoreCLR tools needed by the native runtime build
     - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -subset clr.buildtools $(crossArg) -arch $(archType) -c $(buildConfig) $(officialBuildIdArg) -ci /bl:$(Build.SourcesDirectory)/artifacts/log/clr.buildtools.binlog

--- a/eng/pipelines/libraries/build-job.yml
+++ b/eng/pipelines/libraries/build-job.yml
@@ -79,8 +79,6 @@ jobs:
 
         - ${{ if eq(parameters.isOfficialBuild, true) }}:
           - template: /eng/pipelines/common/restore-internal-tools.yml
-            parameters:
-              installDotnet: true
 
         - script: $(_buildScript)
                 -subset $(_subset)

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "5.0.100-preview.2.20176.6",
+    "version": "5.0.100-preview.4.20202.8",
     "allowPrerelease": true,
     "rollForward": "major"
   },

